File: src/main/java/goodroad/model/gh/GraphHopperResponse.java
package goodroad.model.gh;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class GraphHopperResponse {
    private List<Path> paths;
    private Info info;
}

File: src/main/kotlin/com/GoodRoad/model/gh/GraphHopperResponse.kt
package com.GoodRoad.model.gh

import com.fasterxml.jackson.annotation.JsonProperty

data class GraphHopperResponse(
    val path: List<Path>?,
    val info: Info?
)

data class Path(
    val distance: Double,
    val time: Long,
    val points: String,
    @JsonProperty("points_encoded")
    val pointsEncoded: Boolean = false
)

data class Info(
    val took: Double
)